src/lib/util.ts

```diff
@@ -1,6 +1,31 @@
+import BigNumber from 'bignumber.js'
+
 export function isInteger (value: any) {
-  return typeof value !== 'object'
-    && typeof value !== 'function'
-    && isFinite(value)
-    && Math.floor(value) === value
+  if (BigNumber.isBigNumber(value)) {
+    return value.isFinite()
+      && value.isInteger()
+  } else {
+    return typeof value !== 'object'
+      && typeof value !== 'function'
+      && isFinite(value)
+      && Math.floor(value) === value
+  }
+}
+
+export function bufferToBigNumber (buffer: Buffer): BigNumber {
+  return buffer.reduce(
+    (sum, value) => sum.times(256).plus(value),
```

Comment:
new BigNumber(buffer.toString('hex'), 16) doesn't work?

Reply:
It does. I had that originally and @sharafian thought that was more janky

Reply:
Ugh, lol, why would you listen to a guy who doesn't even drink his vodka and tomato juices :)